include/api_manager/request.h

```diff
@@ -71,6 +71,12 @@ class Request {
   // Sets auth token to the request object. Caller of RequestHandler::Check
   // need to use it compose error message if authentication fails.
   virtual void SetAuthToken(const std::string &auth_token) = 0;
+  // Returns auth token. Used to compose authz_cache key in
+  // check_security_rules.cc. Note that auth_token is set by SetAuthToken in
+  // check_auth.cc. If authentiation in check_auth.cc fails somehow, ESP will
+  // not proceed to check_security_rules.cc. Thus, GetAuthToken is considered to
+  // be safe.
+  virtual std::string GetAuthToken() = 0;
```

Comment:
Change the comment to
this function SHOULD be called after check_auth.   Its implementation is just fetching the token 
in the nginx request ctx which is stored by check_auth.